tests/audio/test_pit.py

```diff
@@ -0,0 +1,210 @@
+    eval_func: str,
+) -> Tuple[Tensor, Tensor]:
+    """a naive implementation of pit based on scipy
+
+    Args:
+        preds: predictions, shape[batch, spk, time]
+        target: targets, shape[batch, spk, time]
+        metric_func: which metric
+        eval_func: min or max
+
+    Returns:
+        best_metric:
+            shape [batch]
+        best_perm:
+            shape [batch, spk]
+    """
+    batch_size, spk_num = target.shape[0:2]
+    metric_mtx = torch.empty((batch_size, spk_num, spk_num), device=target.device)
+    for t in range(spk_num):
+        for e in range(spk_num):
+            metric_mtx[:, t, e] = metric_func(preds[:, e, ...], target[:, t, ...])
+
+    # pit_r = PIT(metric_func, eval_func)(preds, target)
+    metric_mtx = metric_mtx.detach().cpu().numpy()
+    best_metrics = []
+    best_perms = []
+    for b in range(batch_size):
+        row_idx, col_idx = linear_sum_assignment(metric_mtx[b, ...], eval_func == 'max')
+        best_metrics.append(metric_mtx[b, row_idx, col_idx].mean())
+        best_perms.append(col_idx)
+    return torch.from_numpy(np.stack(best_metrics)), torch.from_numpy(np.stack(best_perms))
+
+
+def average_metric(preds: Tensor, target: Tensor, metric_func: Callable) -> Tensor:
+    """average the metric values
+
+    Args:
+        preds: predictions, shape[batch, spk, time]
+        target: targets, shape[batch, spk, time]
+        metric_func: a function which return best_metric and best_perm
+
+    Returns:
+        the average of best_metric
+    """
+    return metric_func(preds, target)[0].mean()
+
+
+snr_pit_scipy = partial(a_naive_implementation_of_pit_based_on_scipy, metric_func=snr, eval_func='max')
+si_sdr_pit_scipy = partial(a_naive_implementation_of_pit_based_on_scipy, metric_func=si_sdr, eval_func='max')
+
+
+@pytest.mark.parametrize(
+    "preds, target, sk_metric, metric_func, eval_func",
+    [
+        (inputs1.preds, inputs1.target, snr_pit_scipy, snr, 'max'),
+        (inputs1.preds, inputs1.target, si_sdr_pit_scipy, si_sdr, 'max'),
+        (inputs2.preds, inputs2.target, snr_pit_scipy, snr, 'max'),
+        (inputs2.preds, inputs2.target, si_sdr_pit_scipy, si_sdr, 'max'),
+    ],
+)
+class TestPIT(MetricTester):
+    atol = 1e-2
+
+    @pytest.mark.parametrize("ddp", [True, False])
+    @pytest.mark.parametrize("dist_sync_on_step", [True, False])
+    def test_pit(self, preds, target, sk_metric, metric_func, eval_func, ddp, dist_sync_on_step):
+        self.run_class_metric_test(
+            ddp,
+            preds,
+            target,
+            PIT,
+            sk_metric=partial(average_metric, metric_func=sk_metric),
+            dist_sync_on_step=dist_sync_on_step,
+            metric_args=dict(metric_func=metric_func, eval_func=eval_func),
+        )
+
+    def test_pit_functional(self, preds, target, sk_metric, metric_func, eval_func):
+        device = 'cuda' if (torch.cuda.is_available() and torch.cuda.device_count() > 0) else 'cpu'
```

Comment:
any reason you cannot use the `self.run_functional_metric_test` method here?

Reply:
done. see https://github.com/PyTorchLightning/metrics/pull/384/commits/a4e347a34793d97c824ebfba922e304af949e27d